File: Python/OOP/Inheritance/Multi-level inehritance (example 1).py
class Institute:
    institute_name = "Institute of Engineering" # just a random name


class Teacher(Institute):
    teacher_name = "Guido van Rossum"
    subject = "Python Programming"


class Student(Teacher):
    student_name = "Abdullah Al Naiem"

    def displayInfo(self):
        print("Institute Name:", self.institute_name)
        print("Teacher Name  :", self.teacher_name)
        print("Student Name  :", self.student_name)
        print("Class of      :", self.subject)


class multiLevelInheritance:
    obj = Student()
    obj.displayInfo()

File: Java/OOP/Inheritance/Multi-level inehritance (example 1).java
class Institute {
    String institute_name = "Institute of Engineering"; // just a random name
}

class Teacher extends Institute {
    String teacher_name = "James Gosling";
    String subject = "Java Programming";
}

class Student extends Teacher {
    String student_name = "Abdullah Al Naiem";

    void displayInfo() {
        System.out.println("Institute Name: " + institute_name);
        System.out.println("Teacher Name  : " + teacher_name);
        System.out.println("Student Name  : " + student_name);
        System.out.println("Class of      : " + subject);
    }
}

class multiLevelInheritance {
    public static void main(String[] args) {
        Student obj = new Student();
        obj.displayInfo();
    }
}